federation/cluster/common.sh

```diff
@@ -74,17 +74,16 @@ function cleanup-federation-api-objects {
   set +o errexit
 
   echo "Cleaning Federation control plane objects"
-  # Delete all resources with the federated-cluster label.
-  $host_kubectl delete pods,svc,rc,deployment,secret -lapp=federated-cluster
-
-  # Delete all PVs bound to PVCs in FEDERATION_NAMESPACE
-  pvs=$($host_kubectl get pvc --namespace=${FEDERATION_NAMESPACE} -o jsonpath='{.items[*].spec.volumeName}')
-  while $host_kubectl delete pv ${pvs} >/dev/null 2>&1; do
-    sleep 2
+  kube::log::status "Removing namespace \"${FEDERATION_NAMESPACE}\" from \"${FEDERATION_KUBE_CONTEXT}\""
+  # Try deleting until the namespace is completely gone.
+  while $host_kubectl --context="${FEDERATION_KUBE_CONTEXT}" delete namespace "${FEDERATION_NAMESPACE}" >/dev/null 2>&1; do
```

Comment:
Do we really need to do this separately at all. Shouldn't the for loop below in line 91 take care of this?

Reply:
@madhusudancs, the for loop below does not include the `federation control plane` context and only iterates over federated clusters. may be some resource are leftover, if we just delete the federation api-server and controllermanager pods, but do not delete the objects in the fcp.

Reply:
I see Ok, fair enough.